pkgs/tools/backup/tsm-client/default.nix

```diff
@@ -74,24 +81,36 @@ let
     '';
   };
 
+  mkSrcUrl = version:
+    let
+      unpacked = lib.trivial.pipe version [
+        (lib.strings.splitString ".")
+        (lib.lists.zipListsWith lib.attrsets.nameValuePair [ "a" "b" "c" "d" ])
+        lib.attrsets.listToAttrs
+      ];
+      inherit (unpacked) a b c d;
+    in
+      "https://public.dhe.ibm.com/storage/tivoli-storage-management/maintenance/client/v${a}r${b}/Linux/LinuxX86_DEB/BA/v${a}${b}${c}/${version}-TIV-TSMBAC-LinuxX86_DEB.tar";
```

Comment:
We can't use lib.versions.majorMinor etc here?

Reply:
Yes, we can.  I didn't know about those -- thanks!  Looks nice now.